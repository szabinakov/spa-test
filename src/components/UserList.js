import React, { useEffect, useState } from "react";
import axios from "axios";
import UserCard from "./UserCard.js";
import Alert from "./Alert.js";
import "../styles/UserList.css";

export default function UserList() {
  const initialState = {
    alert: {
      message: "",
      isSuccess: false,
    },
  };
  const [alert, setAlert] = useState(initialState.alert);
  const [usersForActionList, setUsersForActionList] = useState([]);

  useEffect(() => {
    axios
      .get("https://reqres.in/api/users")
      .then((response) => setUsersForActionList(response.data.data))
      .catch(() => {
        setAlert({
          message: "Something went wrong! Can not load users!",
          isSuccess: false,
        });
      });
  }, []);

  const deleteUser = (id) => {
    axios
      .delete(`https://reqres.in/api/users/${id}`)
      .then(() => {
        setUsersForActionList(
          usersForActionList.filter((user) => user.id !== id)
        );
      })
      .then(() => {
        setAlert({
          message: "User has been successfully removed!",
          isSuccess: true,
        });
      })
      .then(setTimeout(() => setAlert({ message: "", isSuccess: false }), 2000))
      .catch(() => {
        setAlert({
          message: "Can not remove user! Please try again!",
          isSuccess: false,
        });
      });
  };
  return (
    <>
      <Alert message={alert.message} success={alert.isSuccess} />
      <div data-testid="userListContainer" className="userListContainer">
        {usersForActionList.map((user) => (
          <UserCard
            key={user.id}
            name={`${user.first_name} ${user.last_name}`}
            email={user.email}
            avatar={user.avatar}
            deleteUser={deleteUser}
          />
        ))}
      </div>
    </>
  );
}
