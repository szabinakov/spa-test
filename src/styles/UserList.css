.userListContainer {
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  padding-top: 70px;
}

.userCardContainer {
  width: 200px;
  margin: 20px;
  background-color: var(--orange);
  border-radius: 30px;
  margin-bottom: 50px;
  padding: 10px;
  box-shadow: 8px 8px 1px 2px var(--darkblue);
  border: 1px solid var(--darkblue);
}
.userCardContainer:hover {
  box-shadow: 0px 0px 0px;
  transform: translate(10px, 10px);
}
.userCardContainer > .imgContainer {
  text-align: center;
}
.userCardContainer > .imgContainer > img {
  margin-top: 10px;
  border-radius: 50%;
}

.userCardContainer > .nameContainer {
  margin-top: 0px;
  text-align: center;
  font-size: 20px;
  font-weight: bolder;
}

/*****************Delete Button*****************/

.userCardContainer > button {
  align-self: center;
  background-color: var(--orange);
  border: none;
}
.userCardContainer > button > span {
  visibility: hidden;
  width: 120px;
  background-color: rgba(0, 0, 0, 0.274);
  color: #fff;
  text-align: center;
  border-radius: 6px;
  padding: 5px 0;
  margin-left: 15px;
  margin-top: -6px;

  /* Position the tooltip */
  position: absolute;
  z-index: 1;
}

.userCardContainer > button:hover .tooltiptext {
  visibility: visible;
}

/*****************Social Container*****************/

.socialDiv {
  display: flex;
  justify-content: center;
}
.socialIcons {
  padding: 0px;
  margin: 0px;
  list-style: none;
}

.socialIcons li {
  display: inline-block;
  margin: 0.15em;
  position: relative;
}

.socialIcons .iconContainer {
  position: absolute;
  top: 13px;
  left: 13px;
  transition: all 265ms ease-out;
}

.socialIcons a {
  display: inline-block;
  color: white;
}

.socialIcons a:before {
  transform: scale(1);
  -ms-transform: scale(1);
  -webkit-transform: scale(1);
  content: " ";
  width: 40px;
  height: 40px;
  border-radius: 100%;
  display: block;
  background: var(--darkblue);
  transition: all 700ms ease-in-out;
}

.socialIcons a:hover:before {
  transform: scale(0);
  transition: all 400ms ease-in;
}

.socialIcons a:hover .iconContainer {
  transform: scale(2);
  -ms-transform: scale(2);
  -webkit-transform: scale(2);
  color: var(--darkblue);
  transition: all 400ms ease-in;
}

@media only screen and (min-width: 768px) {
  .userCardContainer {
    width: 400px;
    margin: 30px;
  }
  .userCardContainer > .imgContainer {
    width: 150px;
  }
  .userCardContainer > .nameContainer {
    width: 230px;
    text-align: right;
    margin-left: 170px;
    margin-top: -130px;
  }
  .userCardContainer > .emailContainer {
    width: 230px;
    text-align: right;
    margin-left: 170px;
  }
  .userCardContainer > .socialDiv {
    margin-top: 70px;
  }
}

@media only screen and (min-width: 1024px) {
  .userCardContainer {
    margin: 40px;
  }
}
