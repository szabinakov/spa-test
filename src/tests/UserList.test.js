import { render } from "@testing-library/react";
import UserList from "../components/UserList.js";

describe("UserList", () => {
  it("Matches the snapshot", () => {
    const { asFragment } = render(<UserList />);
    expect(asFragment()).toMatchSnapshot();
  });
});
