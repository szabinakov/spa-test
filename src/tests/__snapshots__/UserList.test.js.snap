// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserList Matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="userListContainer"
    data-testid="userListContainer"
  />
</DocumentFragment>
`;
